# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/systemd/no-shared-udev.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- systemd-258.1/meson.build.vanilla	2025-10-27 20:35:30.238188351 +0100
+++ systemd-258.1/meson.build	2025-10-27 20:35:44.275596867 +0100
@@ -2751,7 +2751,7 @@
 
 ############################################################
 
-subdir('rules.d')
+#subdir('rules.d')
 subdir('test')
 
 #####################################################################
--- systemd-258.1/src/udev/meson.build.vanilla	2025-10-27 20:40:16.562332661 +0100
+++ systemd-258.1/src/udev/meson.build	2025-10-27 20:53:44.189884984 +0100
@@ -202,6 +202,7 @@
 ]
 # For the udev alias
 udev_binaries = []
+'''
 foreach binary : udev_binaries_dict
         udev_binaries += binary.get('name')
 endforeach
@@ -263,9 +264,6 @@
                 install_tag : 'udev')
 
 if install_sysconfdir_samples
-        install_data('udev.conf',
-                     install_dir : configfiledir / 'udev',
-                     install_tag : 'udev')
         install_data('iocost/iocost.conf',
                      install_dir : configfiledir / 'udev',
                      install_tag : 'udev')
@@ -282,3 +280,4 @@
 if install_sysconfdir
         install_emptydir(sysconfdir / 'udev/rules.d', install_tag : 'udev')
 endif
+'''
--- systemd-258.1/meson.build.vanilla	2025-10-27 20:55:45.387462175 +0100
+++ systemd-258.1/meson.build	2025-10-27 20:57:55.246131823 +0100
@@ -2433,7 +2433,7 @@
 
 subdir('mime')
 
-alias_target('devel', libsystemd_pc, libudev_pc, systemd_pc, udev_pc)
+alias_target('devel', libsystemd_pc, systemd_pc)
 
 #####################################################################
 
@@ -2942,11 +2942,13 @@
 
 #####################################################################
 
+'''
 udev_targets = []
 foreach bin : udev_binaries
         udev_targets += executables_by_name.get(bin, [])
 endforeach
 alias_target('udev', buildable_rules, udev_targets, udev_units)
+'''
 
 if conf.get('ENABLE_HWDB') == 1
         alias_target('hwdb', auto_suspend_rules, executables_by_name.get('systemd-hwdb'), hwdb_units)
--- systemd-258.1/meson.build.vanilla	2025-10-27 21:12:51.490355114 +0100
+++ systemd-258.1/meson.build	2025-10-27 21:13:16.704617572 +0100
@@ -2159,7 +2159,7 @@
         dependencies : [threads,
                         userspace],
         link_depends : libudev_sym,
-        install : true,
+        install : false,
         install_tag: 'libudev',
         install_dir : libdir)
 
@@ -2174,7 +2174,7 @@
                 libudev_sources,
                 include_directories : includes,
                 implicit_include_directories : false,
-                install : true,
+                install : false,
                 install_tag: 'libudev',
                 install_dir : libdir,
                 link_depends : libudev_sym,
--- systemd-258.1/src/libudev/meson.build.vanilla	2025-10-27 21:10:55.495452605 +0100
+++ systemd-258.1/src/libudev/meson.build	2025-10-27 21:12:30.080149299 +0100
@@ -22,7 +22,7 @@
 libudev_sym = files('libudev.sym')
 libudev_sym_path = libudev_dir_path / 'libudev.sym'
 
-install_headers('libudev.h')
+#install_headers('libudev.h')
 libudev_h_path = libudev_dir_path / 'libudev.h'
 
 libudev_basic = static_library(
@@ -41,6 +42,6 @@
         input : 'libudev.pc.in',
         output : 'libudev.pc',
         command : [jinja2_cmdline, '@INPUT@', '@OUTPUT@'],
-        install : pkgconfiglibdir != 'no',
+        install : false,
         install_tag : 'devel',
         install_dir : pkgconfiglibdir)

